File: src/main/java/com/todo/javafullstack/models/Todo.java
package com.todo.javafullstack.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;

import java.time.LocalDate;

/***
 * Todo Object using lombok annotations to increase readability and fewer
 * boiler plate code.
 */

@Getter
@Setter
@Component
//@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Todo {

//    @Id
//    @GeneratedValue
//    private long id;
    private String title;
    private String description;
    private LocalDate dueDate;
    private LocalDate completionDate;

    @Override
    public String toString(){
        return "Employee{title="+title+",description="+description+",dueDate="+dueDate+",completionDate="+completionDate+"}";
    }
}

File: todo-ui/src/app/models/todo.ts
export class Todo {
    title: string;
    description: string;
    dueDate: Date;
    completionDate: Date;

constructor(    title: string,
                description: string,
                dueDate: Date,
                completionDate: Date,) {
    this.title = title;
    this.description = description;
    this.dueDate = dueDate;
    this.completionDate = completionDate;
}

static objectToTodo(object: any) {
    let todo = new Todo(object.title, object.description, object.dueDate, object.completionDate);
    return todo;
  }
}
